src/interlocking_component/router.py

```diff
@@ -1,12 +1,51 @@
+from typing import List
+
+from src.wrapper.simulation_objects import Node, Track
+
+
 class Router:
     """This class calculates routes for trains from track segment to track segment."""
 
-    def get_route(self, start_track_segment: "Track", end_track_segment: "Track"):
+    def get_route(self, start_track: Track, end_track: Track) -> List[Node]:
         """This method returns a route from a start track_segment to an end track_segment.
+        It returns a list of Sumo Nodes from the Signal the train is currently driving
+        toward to the Node right after the end track.
 
-        :param start_track_segment: The track segment where the route will begin.
-        :type start_track_segment: Track
-        :param end_track_segment_id: The track segment where the route will end.
-        :type end_track_segment: Track
+        :param start_track: The track segment where the route will begin.
+        :type start_track: Track
+        :param end_track: The track segment where the route will end.
+        :type end_track: Track
+        :return: list of Nodes from the Signal after the start track
+        to the Node right after the end track
+        :rtype: List[Node]
         """
-        raise NotImplementedError()
+        start_node = start_track.to_node
+        penultimate_node = end_track.from_node
+        # The next part is dijkstra as a first mvp.
+        # This will lead to deadlocks if two trains drive in opposite directions.
+        distances = {}
+        previous_nodes = {}
+        distances[start_node] = 0
+        current_index = 0
+        current_node = start_node
+        while True:
+            sorted_distances = sorted(distances.items(), key=lambda item: item[1])
```

Comment:
What does `item[1]` contain? Could you please add a small comment about the sorting or replace the `1` with an semantic constant 👍

Reply:
```distances``` is a dict, so each ```item``` is one entry in the dict. ```item[1]``` then referes to the value of the dict entry. So this sorts the dict based on the values. I will add a comment here.

Reply:
But what is `item[0]`? :D A route? A number? 🙃